File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        StringBuilder result = new StringBuilder();
        Stack<Character> stack = new Stack<>();
        // System.out.println(s.toCharArray()); leet**cod*e
        for(char ch : s.toCharArray()){
            if(ch == '*') {
                stack.pop();
            } else {
                stack.push(ch);
            }
        }
        // System.out.println(stack);[l, e, c, o, e]
        while(!stack.isEmpty()){
            result.append(stack.pop());
            // System.out.println(result); eocel
        }
        return result.reverse().toString();
    }
}

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = []
        stri = ""
        for i in range(len(s)):
            while stack and stack[-1] == '*':
                stack.pop()
                stack.pop()
            stack.append(s[i])
            if stack[-1] == "*":
                stack.pop()
                stack.pop()

        for i in stack:
            stri+=i
        return stri

        